lms/djangoapps/instructor/tests/test_api.py

```diff
@@ -3580,3 +3611,110 @@ def test_success_carriage_return_line_feed(self, mock_store_upload, mock_cohort_
         self.verify_success_on_file_content(
             'username,email,cohort\r\nfoo_username,bar_email,baz_cohort', mock_store_upload, mock_cohort_task
         )
+
+
+@override_settings(MODULESTORE=TEST_DATA_MOCK_MODULESTORE)
+class TestAddInvoiceTransactionPaymentAndRefund(ModuleStoreTestCase):
+    """
+    Test data dumps for Invoice Transactions
+    """
+
+    def setUp(self):
+        """
+        Fixtures.
+        """
+        self.course = CourseFactory.create()
+        CourseModeFactory.create(course_id=self.course.id, min_price=50)
+        self.instructor = InstructorFactory(course_key=self.course.id)
+        self.client.login(username=self.instructor.username, password='test')
+        CourseSalesAdminRole(self.course.id).add_users(self.instructor)
+
+        self.sale_invoice_1 = Invoice.objects.create(
+            total_amount=1234.32, company_name='Test1', company_contact_name='TestName',
+            company_contact_email='Test@company.com', recipient_name='Testw', recipient_email='test1@test.com',
+            customer_reference_number='2Fwe23S', internal_reference="A", is_valid=True, course_id=self.course.id
+        )
+        self.invoice_item = CourseRegistrationCodeInvoiceItem.objects.create(
+            invoice=self.sale_invoice_1,
+            qty=1,
+            unit_price=1234.32,
+            course_id=self.course.id
+        )
+        self.generate_code_url = reverse(
+            'make_invoice_transaction', kwargs={'course_id': self.course.id.to_deprecated_string()}
+        )
+
+        data = {
+            'invoice_id': self.sale_invoice_1.id, 'amount': 110, 'amount_type': 'payment',
+            'comments': 'testing comments'
+        }
+
+        response = self.client.post(self.generate_code_url, data, **{'HTTP_HOST': 'localhost'})
+        self.assertEqual(response.status_code, 200, response.content)
+
+    def test_add_new_invoice_transaction_payment_with_invalid_invoice_id(self):
+        """
+        Test to add invoice transaction with invalid invoice id
+        """
```

Comment:
Style nitpick: our test runner displays these docstrings instead of the test method in its output, which can make it hard to find the original test if it fails.  For this reason, we usually don't include docstrings in `test_` methods.